panos_upgrade_assurance/check_firewall.py

```diff
@@ -870,6 +864,185 @@ def check_ssl_cert_requirements(self, rsa: dict = {}, ecdsa: dict = {}) -> Check
+                details = "hourly"
+            elif schedule_type.split("-")[1].isnumeric():
+                time_distance = int(schedule_type.split("-")[1])
+                details = f"every {time_distance} minutes"
+            else:
+                raise exceptions.MalformedResponseException(f"Unknown schedule type: {schedule_type}.")
+        elif schedule_type == "real-time":
+            details = "unpredictable (real-time)"
+        else:
+            raise exceptions.MalformedResponseException(f"Unknown schedule type: {schedule_type}.")
+
+        return time_distance, details
+
+    def check_scheduled_updates(self, test_window: int = 60) -> CheckResult:
+        """Check if any Dynamic Update job is scheduled to run within the specified time window.
+
+        # Parameters
+
+        test_window (int, optional): (defaults to 60 minutes). A time window in minutes to look for an update job occurrence.
+            Has to be a value between `60` and `10080` (1 week equivalent). The time window is calculated based on the device's
+            local time (taken from the management plane).
+
+        # Raises
+
+        MalformedResponseException: Thrown in case API response does not meet expectations.
+
+        # Returns
+
+        CheckResult: Object of [`CheckResult`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkresult) class taking \
+            value of:
+
+        * [`CheckStatus.SUCCESS`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) when there is no update job
+            planned within the test time window.
+        * [`CheckStatus.FAIL`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) otherwise, `CheckResult.reason`
+            field contains information about the planned jobs with next occurrence time provided if possible.
+        * [`CheckStatus.ERROR`](/panos/docs/panos-upgrade-assurance/api/utils#class-checkstatus) when the `test_window` parameter
+            does not meet criteria.
+
+        """
+        if not isinstance(test_window, int):
+            raise exceptions.WrongDataTypeException(
+                f"The test_windows parameter should be of type <class int>, got {type(test_window)} instead."
+            )
+
+        result = CheckResult()
+
+        self._mp_now = self._node.get_mp_clock()
+
+        schedules = self._node.get_update_schedules()
+        if not schedules:
+            result.status = CheckStatus.SKIPPED
+            result.reason = "No scheduled job present on the device."
+            return result
+
+        if test_window < 60:
+            result.status = CheckStatus.ERROR
+            result.reason = "Schedules test window is below the supported, safe minimum of 60 minutes."
+            return result
+        if test_window > 10080:
+            result.status = CheckStatus.ERROR
+            result.reason = "Schedules test window is set to over 1 week. This test will always fail."
+            return result
```

Comment:
I think we can merge this if condition with the above one. Something like:

```suggestion
        if test_window < 60 or test_window > 10080:
            result.status = CheckStatus.ERROR
            result.reason = f"Schedules test window is not in supported scope (should be between 60 - 10080), found {test_window}."
            return result
```

Reply:
well, you compress the code, but you loose the description, which is explaining the assumption done when developing this test, you know the min-max values, but you do not know why they are as they are.
And I'm guessing that a network engineer or anyone (this includes myself) running this test will not study pan.dev before running the play to find out why it `60< value > 10080` :)